src/arithmetic/entity_funcs/entity_funcs.c

```diff
@@ -34,6 +35,22 @@ SIValue AR_LABELS(SIValue *argv, int argc) {
 	return SI_ConstStringVal(label);
 }
 
+/* returns true if node contains header, otherwise false. */
+SIValue AR_HAS_LABELS(SIValue *argv, int argc) {
+	if(SI_TYPE(argv[0]) == T_NULL) return SI_NullVal();
+
+	char *label = "";
+	Node *node = argv[0].ptrval;
+	SIValue labels = argv[1];
+	if(SIArray_Length(labels) > 1) return SI_BoolVal(false);
+	
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+	Graph *g = gc->g;
+	int labelID = Graph_GetNodeLabel(g, ENTITY_GET_ID(node));
```

Comment:
This can be expensive (scanning through each label matrix)
Considering the alternative:
Going from given `label` to `schema`, then validating that `labels_matrices[schema->id] [ENTITY_GET_ID(node), ENTITY_GET_ID(node)]` is set.